metric.go

```diff
@@ -244,6 +244,46 @@ func SanitizeNameAsTagValue(name string) string {
 	return ""
 }
 
+// EatDots removes multiple consecutive, leading, and trailing dots
+// from name. If the provided name is only dots, it will return an
+// empty string
+func EatDots(name string) string {
+	if len(name) == 0 {
+		return ""
+	}
+
+	n := strings.Trim(name, ".")
```

Comment:
I think we need to keep in mind that MT / CRNG are probably going to call this very often. so personally i'd write that without using `strings.Trim`... can we benchmark the difference between this version and one which simply iterates over the chars of `name` and does a `name = append(name[:i], name[i+1:]...)` for every double dot? i think the simple iterator would be more efficient

Reply:
I pushed a commit which improves it:
```
replay@mst-nb:~/go/src/github.com/raintank/schema$ benchcmp bench.old bench.new
benchmark              old ns/op     new ns/op     delta
BenchmarkEatDots-8     1127          592           -47.47%

benchmark              old allocs     new allocs     delta
BenchmarkEatDots-8     26             13             -50.00%

benchmark              old bytes     new bytes     delta
BenchmarkEatDots-8     739           323           -56.29%
```